app/src/main/java/com/android/sample/model/profile/ProfileViewModel.kt

```diff
@@ -55,6 +55,22 @@ open class ProfileViewModel @Inject constructor(private val repository: Profiles
         onFailure = {})
   }
 
+  fun addLikedActivity(userId: String, activityId: String) {
+    repository.addLikedActivity(
+        userId = userId,
+        activityId = activityId,
+        onSuccess = { fetchUserData(userId) },
+        onFailure = {})
```

Comment:
The onFailure callback is currently empty. . Consider logging these errors or showing a message to the user.